modules/customizing-the-login-page.adoc

```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// * web_console/customizing-the-web-console.adoc
+
+[id="customizing-the-login-page_{context}"]
+= Customizing the login page
+
+Create Terms of Service information with custom login pages. Custom login pages
+can also be helpful if you use a third-party login provider, like GitHub or
+Google, to show users a branded page that they trust and expect before being
+redirected to the authentication provider. You can also render custom error
+pages during the authentication process.
+
+.Prerequisites
+
+* You must have administrator privileges.
+
+.Procedure
+
+. Run the following commands to create templates you can modify:
++
+----
+$ oc adm create-login-template > login-template.html
+$ oc adm create-provider-selection-template > provider-selection-template.html
+----
+
+. To use your custom login page, provider selection page, or custom error page,
+set the following options:
```

Comment:
You'll need to create the secret with the page as well.

```
$ oc adm create-login-template > login.html
$ oc adm create-provider-selection-template > providers.html
$ oc create secret generic login-template --from-file=login.html -n openshift-config
$ oc create secret generic providers-template --from-file=providers.html -n openshift-config
```

Then

```
$ oc edit oauths cluster
```

```yaml
spec:
  templates:
    login:
        name: login-template
    providerSelection:
        name: providers-template
```

See

```
$ oc explain oauths.spec.templates
KIND:     OAuth
VERSION:  config.openshift.io/v1

RESOURCE: templates <Object>

DESCRIPTION:
     templates allow you to customize pages like the login page.

FIELDS:
   error        <Object>
     error is the name of a secret that specifies a go template to use to render
     error pages during the authentication or grant flow. The key "errors.html"
     is used to locate the template data. If specified and the secret or
     expected key is not found, the default error page is used. If the specified
     template is not valid, the default error page is used. If unspecified, the
     default error page is used. The namespace for this secret is
     openshift-config.

   login        <Object>
     login is the name of a secret that specifies a go template to use to render
     the login page. The key "login.html" is used to locate the template data.
     If specified and the secret or expected key is not found, the default login
     page is used. If the specified template is not valid, the default login
     page is used. If unspecified, the default login page is used. The namespace
     for this secret is openshift-config.

   providerSelection    <Object>
     providerSelection is the name of a secret that specifies a go template to
     use to render the provider selection page. The key "providers.html" is used
     to locate the template data. If specified and the secret or expected key is
     not found, the default provider selection page is used. If the specified
     template is not valid, the default provider selection page is used. If
     unspecified, the default provider selection page is used. The namespace for
     this secret is openshift-config.
```

Reply:
👍